### PR TITLE
Handle Multipart/mixed Bounces and Auto-submitted Messages

### PR DESCRIPTION
**Pull request by eshryane**
Look for automated responses in plaintext mail messages and do not handle them as Whois updates.

Multipart/mixed messages may also contain a delivery failure report.

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/AutoSubmittedMessageParser.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+@Component
+public class AutoSubmittedMessageParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AutoSubmittedMessageParser.class);
+
+    private final boolean enabled;
+
+    @Autowired
+    public AutoSubmittedMessageParser(@Value("${mail.smtp.from:}") final String smtpFrom) {
+        this.enabled = ! Strings.isNullOrEmpty(smtpFrom);
```
Comment (MiguelAHM) [inline whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/AutoSubmittedMessageParser.java:28]: white space between ! and Strings?

File: whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java [hunk 2]
```diff
@@ -45,35 +48,100 @@ public BouncedMessageParser(@Value("${mail.smtp.from:}") final String smtpFrom)
...
+                    }
+                }
+            } catch (MessagingException | IOException | IllegalStateException ex){
+                throw new MailParsingException("Error parsing multipart report");
+            }
+            // multipart report can *only* be a failure
+            throw new MailParsingException("MultiPart message without failure report");
+        }
+
+        if (isMultipartMixed(message)) {
```
Comment (MiguelAHM) [inline whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java:74]: In this case. Shouldn't we return a MailParsingException??
I mean, should multipartMixed be processed as an error?? as we do with multipart report??

The mail updates are always in text plain, right? So everything that is not a text plain should not go to mail updates

File: whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java [hunk 3]
```diff
@@ -45,35 +48,100 @@ public BouncedMessageParser(@Value("${mail.smtp.from:}") final String smtpFrom)
...
-                final DeliveryStatus deliveryStatus = deliveryStatus(message);
-                if (isFailed(deliveryStatus)) {
-                    final MimeMessage returnedMessage = multipartReport.getReturnedMessage();
-                    final String messageId = getMessageId(returnedMessage.getMessageID());
-                    final List<String> recipient = extractRecipients(deliveryStatus);
-                    return new EmailMessageInfo(recipient, messageId);
+        if (isMultipartReport(message)) {
+            try {
+                final MultipartReport multipartReport = multipartReport(message.getContent());
+                if (isReportDeliveryStatus(multipartReport)) {
```
Comment (MiguelAHM) [inline whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java:58]: Line 58 - 66 and 77 -85 are the same
can we extract this into a method called something like "extractMailMessageInfo"?

File: whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java [hunk 4]
```diff
@@ -45,35 +48,100 @@ public BouncedMessageParser(@Value("${mail.smtp.from:}") final String smtpFrom)
...
+                LOGGER.error(String.format("%s: %s", ex.getClass().getName(), ex.getMessage()), ex);
+                throw new MailParsingException("Error parsing multipart report");
             }
-        } catch (MessagingException | IOException | IllegalStateException ex){
-            throw new MailParsingException("Error parsing multipart report");
         }
-        throw new MailParsingException("MultiPart message without failure report");
+
+        // fall through: message is not bounced message
+        return null;
```
Comment (MiguelAHM) [inline whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java:93]: Maybe early check that it is a plain text and return null if it a plain text (at the beginning of the method?)
and return a MailParsingException here...this is more strict but we would just allow mail updates when text plain

File: whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java:74]: The mail updates are always in text plain, right? -> NO

File: whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/AutoSubmittedMessageParser.java (hunk 1 above)
Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/AutoSubmittedMessageParser.java]: Fixed

File: whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java (hunk 3 above)
Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java:60]: I don't want to disturb the existing code, and they are a bit different still - we throw an exception in the multipart/report code that we don't throw in the multipart/mixed code.
I added a TODO for now.

**Comment by eshryane** [comment]
Integration tests passed

**Pull request merged**